apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts

```diff
@@ -0,0 +1,688 @@
+    const group_name = button.data('group-name');
+    const uid_list = button.data('uid-list');
+    const date_formatted = button.data('date-formatted');
+    const score_perc = button.data('score-perc');
+    const assessment_instance_id = button.data('assessment-instance-id');
+    const modal = $(this);
+
+    modal.find('form').on('submit', (e) => {
+      e.preventDefault();
+      $.post(
+        $(e.target).attr('action') ?? '',
+        $(e.target).serialize(),
+        function () {
+          refreshTable();
+        },
+        'json',
+      );
+      modal.modal('hide');
+    });
+
+    modal.find('.modal-uid').text(uid);
+    modal.find('.modal-name').text(name);
+    modal.find('.modal-group-name').text(group_name);
+    modal.find('.modal-uid-list').text(uid_list);
+    modal.find('.modal-number').text(number);
+    modal.find('.modal-date').text(date_formatted);
+    modal.find('.modal-score-perc').text(score_perc);
+    modal.find('.modal-assessment-instance-id').val(assessment_instance_id);
+  });
+
+  $('#deleteAllAssessmentInstancesModal').on('show.bs.modal', function () {
+    const modal = $(this);
+
+    modal.find('form').on('submit', (e) => {
+      e.preventDefault();
+      $.post(
+        $(e.target).attr('action') ?? '',
+        $(e.target).serialize(),
+        function () {
+          refreshTable();
+        },
+        'json',
+      );
+      modal.modal('hide');
+    });
+  });
+
+  $('[data-toggle="modal"]').on('click', function (e) {
+    e.stopPropagation(); // Keep click from changing sort
+    $($(e.currentTarget).data('target')).modal('show');
+  });
+
+  function tableColumns(assessmentGroupWork: boolean) {
+    const assessmentInstanceIdColumn = {
+      field: 'assessment_instance_id',
+      title: '<span class="sr-only">Assessment Instance</span>',
+      sortable: true,
+      sorter: detailsLinkSorter,
+      formatter: detailsLinkFormatter,
+      class: 'align-middle sticky-column text-nowrap',
+      switchable: false,
+    };
+    const nonspecificColumns = [
+      {
+        field: 'number',
+        title: 'Instance',
+        visible: false,
+        sortable: true,
+        class: 'text-center align-middle',
+        switchable: true,
+      },
+      {
+        field: 'score_perc',
+        title: 'Score',
+        sortable: true,
+        class: 'text-center align-middle',
+        formatter: scorebarFormatter,
+        switchable: true,
+        searchFormatter: false,
+      },
+      {
+        field: 'date_formatted',
+        title: 'Date started',
+        sortable: true,
+        sortName: 'date',
+        class: 'text-center align-middle text-nowrap',
+        switchable: true,
+      },
+      {
+        field: 'duration',
+        title:
+          'Duration <button class="btn btn-xs" type="button" title="Show duration help" data-toggle="modal" data-target="#duration-help"><i class="bi-question-circle-fill" aria-hidden="true"></i></button>',
```

Comment:
Consider using `html` tagged-template literals here and elsewhere in this file when you're constructing HTML as a string, e.g.:

```suggestion
          html`Duration <button class="btn btn-xs" type="button" title="Show duration help" data-toggle="modal" data-target="#duration-help"><i class="bi-question-circle-fill" aria-hidden="true"></i></button>`.toString(),
```

Prettier will of course fix that up to look nicer 🙂

Reply:
Updated locally here and elsewhere in [306ccbe](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/306ccbe9c0ebd4481747d7e7ddce945db6fff328).